File: removeDuplicatesSorted.py
#Time Complexity : O(n) at every index their is the movement of slow and faster pointer
#Space Complexity : O(1) result is return in form of an integer
#Did this code successfully run on Leetcode : Yes
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        #for null values
        if nums== None:
            return 0
        #defining the slow pointer and fast pointer work will be done by loop itself
        slow =1 
        count =1
        #for loop for removing the duplicacy and it will work like a faster point to iterate over each element 
        for i in range(1,len(nums)):
            if nums[i]==nums[i-1]:
                #increase the count once come across the same number
                count += 1

            else:
                #reseting the count each time new number is encountered
                count =1
            if count <=2:
                #the elements will get overwrite to remove the duplicacy of unneccessary numbers in the array
                nums[slow]= nums[i]

                #slow will get increment for each element with element <= k times
                slow +=1
 
        return slow

File: RemoveDuplicatesSorted.java

// Time Complexity : O(n)
// Space Complexity : O(1)
// Did this code successfully run on Leetcode : yes
// Three line explanation of solution in plain english: No

// Your code here along with comments explaining your approach
/*1. Keep a pointer i and j and check if the two consecutive numbers are same and keep a counter to increment if duplicates are found.
copy the value of i to j. Increment i at each iteartion.
Run the loop until the length of nums * 
 */

class Solution {
    public int removeDuplicates(int[] nums) {
        //base case 
        if(nums.length<2) return nums.length;
        int n =nums.length, j=1, count = 1;
        for(int i=1;i<nums.length;i++){
            if(nums[i]==nums[i-1]){
                count++;
            }
            else{
                count = 1;
            }
            if(count<=2){
                nums[j] = nums[i];
                j++;
            }
        }
        return j;
        }
         
    }